nlp/build-model-worker-on-demand/src/main/kotlin/WorkerOnDemandVerticle.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2017/2020 e-voyageurs technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.tock.nlp.build.ondemand
+
+import io.vertx.core.AbstractVerticle
+import io.vertx.core.Handler
+import mu.KotlinLogging
+import java.time.LocalTime
+import java.util.concurrent.TimeUnit
+
+class WorkerOnDemandVerticle(
+    private val workerOnDemandType: WorkerOnDemandType,
+    private val buildType: String,
+    private val delayBetweenJob: Long,
+    private val timeFrame: List<Int>
+) : AbstractVerticle() {
+
+    private val logger = KotlinLogging.logger {}
+
+    private var handler: Handler<Long>? = null
+
+    private var workerOnDemand: WorkerOnDemand? = null
+
+    override fun start() {
+        logger.info("Starting WorkerOnDemandVerticle for $buildType ...")
+
+        workerOnDemand = WorkerOnDemandProvider.provide(
+            type = workerOnDemandType,
+            properties = mapOf("TOCK_BUILD_TYPE" to buildType) + workerProperties()
+        )?.apply {
+            logger.info { "WorkerOnDemand ${this@WorkerOnDemandVerticle.javaClass.simpleName} loaded for $buildType" }
+            handler = Handler {
+                if (LocalTime.now().isInTimeFrame()) {
```

Comment:
is time zone handled by this method? I think LocalTime will return the time defined in the machine where the worker is running which can have a different time zone with the user